Include information about tool maintenance

ChrisJBurns opened the pull request:
- Include information about how to maintain the tool when there is a topic increase

Commit:
Include information about tool maintenance

- Include information about how to maintain the tool when there is a topic increase

Commit:
Add README.md for schema changes

- Added README.md to document what needs to happen if schema changes are made

Commit:
Change code snippet format

- Remove json from code snippet format as the snippet is not correct json

Commit:
Update links in README.md

- Include link to `README.md` in `/schemas`
- Include link to `/schemas`

Commit:
Correct spelling mistakes in README.md

- Correct spelling mistakes

jsimons1989 commented on schemas/README.md:
```diff
@@ -0,0 +1,51 @@
+# Maintenance
+
+## Introduction
+At some point in time there will be an increase an topics or even a change to the existing topic structure. If at any point one of these happen, this will require a change to the Avro schemas in further requiring a change in the structs that represent the Avro schemas. These structs are used by the kafka-management-tool in order to map the unmarshalled JSON message from with the specified topic.
```
These doesn't look correct:

- "an increase an topics"
- "JSON message from with the specified topic"

ChrisJBurns replied:
ahh yes, meant to be `in`... good spot

jsimons1989 commented on schemas/README.md:
```diff
@@ -0,0 +1,51 @@
+
+### Addition of new topic and Avro schema
+If a new topic is to be added, there will need to be a new Avro schema file that represents the message structure. Additionally, the services that produce and consume messages from the new topic will need to create the mapping object/struct that represents the topic schema. In `Java` services this is a `Pojo` and in `Go`, a `struct`. For information on how the struct will can be created for the k ...
+Additionally, once the new struct has been created, another condition will need to be added to the `schemas.go` file to make sure that the correct struct is used when the new topic is specified in the tool.
```
I would find this easier to read if the example was with the explanation.

ChrisJBurns replied:
Done

Commit:
Correct spelling mistake

- Correct spelling mistake of the word `an` to `in`

Commit:
Put examples in the correct section

- Moved examples from a separate section and put them within the specific scenarios for a more fluid understanding

Commit:
Spelling correction

jsimons1989 commented on schemas/README.md:
```diff
@@ -0,0 +1,74 @@
+If a change is made to an existing topic structure/schema or if a new topic and Avro schema has been added this will need to be reflected in the kafka-management-tool schema struct for the corresponding Avro schema.
+
+### Addition of new topic and Avro schema
+If a new topic is to be added, there will need to be a new Avro schema file that represents the message structure. Additionally, the services that produce and consume messages from the new topic will need to create the mapping object/struct that represents the topic schema. In `Java` services this is a `Pojo` and in `Go`, a `struct`. For information on how the struct will can be created for the k ...
```
Needs a slight rewording: "how the struct will can be created"

jsimons1989 commented on schemas/README.md:
````diff
@@ -0,0 +1,74 @@
+The `Go` struct will look like the following
+
+```go
+type TxClosed struct {
````
The example above mentions "tx-open" topic but you have called the struct "TxClosed"

jsimons1989 commented on schemas/README.md:
````diff
@@ -0,0 +1,74 @@
+```go
+type TxClosed struct {
+	Timedate                string  `avro:"timedate"                            json:"timedate"`
+	TransactionDescription  string  `avro:"transaction_details"                 json:"transaction_details"`
````
I think you should be consistent here and call the variable TransactionDetails

jsimons1989 commented on schemas/README.md:
````diff
@@ -0,0 +1,74 @@
+}
+```
+
+Additionally in the Go structs and/or Java pojos, there would be an extra field added to it to reflect the new field in the Avro schema.
````
I think this README would be simpler if it only referenced the changes in this repository. It's a bit confusing after this example to mention "in the Go structs and/or Java pojos" when this repo has no Java POJOs and not have an example of the changes to the TxClosed struct.
